torch/export/dynamic_shapes.py

```diff
@@ -420,31 +421,28 @@ def check_same_bounds(dim):
         else:
             bounds[dim.__name__] = (dim.min, dim.max)
 
+    def update_single_symbol(dim, tensor, i):
+        if isinstance(dim, int):
+            dim = Dim(name=f"explicit_static_dim_{num_explicit_static_dims[0]}", min=dim, max=dim)
```

Comment:
Interesting approach. While this should work, I'd prefer some kind of subclass of `Dim` here.

Also, the name has to be unique, there is no guarantee that the name here will not conflict.